src/pages/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  FC,
+  memo,
+  useEffect,
+  useState,
+} from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Loader } from '../../components/Loader';
+import { Person } from '../../types/Person';
+import { PeopleTable } from '../../components/PeopleTable';
+import { preparePeople } from '../../utils/preparePeople';
+import { PeopleFilters } from '../../components/PeopleFilters';
+import { filterPeopleBySex } from '../../utils/filterPeopleBySex';
+import { filterPeopleByQuery } from '../../utils/filterPeopleByQuery';
+import { filterPeopleByCenturies } from '../../utils/filterPeopleByCenturies';
+import { sortPeople } from '../../utils/sortPeope';
+
+export const PeoplePage: FC = memo(() => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [
+    filteredAndSortedPeople,
+    setFilteredAndSortedPeople,
+  ] = useState<Person[]>([]);
+  const [isPeopleLoading, setIsPeopleLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const { selectedPersonSlug = '0' } = useParams();
+  const [searchParams] = useSearchParams();
+
+  const loadPeople = async () => {
+    try {
+      const peopleFromServer = await getPeople();
+      const preparedPeopleFromServer = preparePeople(peopleFromServer);
+
+      setFilteredAndSortedPeople(preparedPeopleFromServer);
+      setPeople(preparedPeopleFromServer);
+    } catch {
+      setError(true);
+    } finally {
+      setIsPeopleLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  const params = {
+    sex: searchParams.get('sex'),
+    query: searchParams.get('query'),
+    centuries: searchParams.getAll('centuries'),
+    sort: searchParams.get('sort'),
+    order: searchParams.get('order'),
+  };
+
+  useEffect(() => {
+    let currentFilteredPeople = [...people];
+
+    if (params.sex) {
+      currentFilteredPeople = filterPeopleBySex(
+        currentFilteredPeople,
+        params.sex,
+      );
+    }
+
+    if (params.query) {
+      currentFilteredPeople = filterPeopleByQuery(
+        currentFilteredPeople,
+        params.query,
+      );
+    }
+
+    if (params.centuries.length) {
+      currentFilteredPeople = filterPeopleByCenturies(
+        currentFilteredPeople,
+        params.centuries,
+      );
+    }
```

Comment:
will be better to move the logic of filtering inside some general filter function instead of processing it directly in component